ios/FluentUI/Tab Bar/SideTabBar.swift

```diff
@@ -0,0 +1,299 @@
+    @objc optional func sideTabBar(_ sideTabBar: SideTabBar, didActivate avatarView: AvatarView)
+}
+
+/// View for a vertical side tab bar that can be used for app navigation.
+/// Optimized for horizontal regular + vertical regular size class configuration. Prefer using TabBarView for other size class configurations.
+@objc(MSFSideTabBar)
+open class SideTabBar: UIView {
+    /// Delegate to handle user interactions in the side tab bar.
+    @objc public weak var delegate: SideTabBarDelegate?
+
+    /// The avatar view that displays above the top tab bar items.
+    /// The avatar view's size class should be AvatarSize.medium.
+    @objc open var avatarView: AvatarView? {
+        willSet {
+            avatarView?.removeGestureRecognizer(avatarViewGestureRecognizer)
+            avatarView?.removeFromSuperview()
+        }
+        didSet {
+            if let view = avatarView {
+                view.translatesAutoresizingMaskIntoConstraints = false
+                addSubview(view)
+
+                view.addGestureRecognizer(avatarViewGestureRecognizer)
+            }
+
+            setupLayoutConstraints()
+        }
+    }
+
+    /// Tab bar iems to display in the top section of the side tab bar.
+    /// These TabBarItems don't need landscape images.
+    @objc open var topItems: [TabBarItem] = [] {
+        willSet {
+            willSetItems(in: .top)
+        }
+        didSet {
+            didSetItems(in: .top)
+        }
+    }
+
+    /// Tab bar iems to display in the bottom section of the side tab bar.
+    /// These items do not have a selected state.
+    /// These TabBarItems don't need landscape images.
+    @objc open var bottomItems: [TabBarItem] = [] {
+        willSet {
+            willSetItems(in: .bottom)
+        }
+        didSet {
+            didSetItems(in: .bottom)
+        }
+    }
+
+    /// Selected tab bar item in the top section of the side tab bar.
+    @objc open var selectedTopItem: TabBarItem? {
+        willSet {
+            if let item = selectedTopItem {
+                itemView(with: item, in: .top)?.isSelected = false
+            }
+        }
+        didSet {
+            if let item = selectedTopItem {
+                itemView(with: item, in: .top)?.isSelected = true
+            }
+        }
+    }
+
+    @objc public override init(frame: CGRect) {
+        super.init(frame: frame)
+        initialize()
+    }
+
+    @available(*, unavailable)
+    @objc public required init?(coder: NSCoder) {
+        preconditionFailure("init(coder:) has not been implemented")
+    }
+
+    /// Helper method to insert the view in a superview.
+    @objc public func insert(in superView: UIView) {
+        superView.addSubview(self)
+
+        NSLayoutConstraint.activate([
+            leadingAnchor.constraint(equalTo: superView.leadingAnchor),
+            topAnchor.constraint(equalTo: superView.topAnchor),
+            bottomAnchor.constraint(equalTo: superView.bottomAnchor)
+        ])
+    }
```

Comment:
This pattern still seems a bit odd to me. Why not make the public API a SideTabBarViewController so we aren't having this view take implicit ownership/knowledge of its superview?

Reply:
This is just a helper method to insert the view in its superview and position it along the leading edge.
I added a comment in the methods header to point that out.
The expectation is that, in most cases, this is how the view will get inserted in a view hierarchy so we can make it easier for consumers of this view.

Reply:
I guess a rephrasing is - is there any reason not to just expose a view controller instead of a view if the designed usage is as a decorator for a contentViewController?

Reply:
I was inspired by TabBarView which is just a view.
This control only contains view logic: it lays out subviews and intercepts touch events to pass them to a delegate. It could be a view controller, but I don't see any benefit. This class doesn't take care of any presentation logic, there is no controller logic pushing model data into views (MVC), there is no need for managing the lifecycle of the view, etc. 
I'd prefer to remove this helper method and make it clear that this is just a view that can be positioned anywhere.

Reply:
Yeah the point about being aligned with TabBarView makes sense but makes me nervous at the same time. Some thoughts:

- Fluent design does not really state that it can be positioned anywhere. It is part of fluent design that the TabBarView used to be restricted to the bottom of the screen and this control is being added because fluent design now says that a similar but different control can be used for the leading edge of the screen
- View controllers by nature don't take care of presentation logic of themselves, unless you're referring to the presentation of the content view controller. If that, I do kinda think it should take care of that presentation logic. If anyone is using this control, per fluent design the tabs should not execute actions but instead should swap out contentViewControllers.
- I guess the ultimate question - how is this control different than something like UIPageViewController or UITabBarController, both of which Apple chose UIViewControllers for? Default should be to follow Apple's patterns and idioms, so what is the distinction here that makes that undesirable?

Reply:
I see your point. The believe a good solution here would be to make SideTabBarView private and expose a view controller that manages the SideTabBarView and swaps view controllers.
It was my intention to build this view controller to start with because, depending on the window size class, we will need to alternate between SideTabBarView and TabBarView.
This view controller will involve a lot more work.

Reply:
So proposal is to check in as is and then add a view controller on top of it? That seems reasonable to me, but maybe we should start SideBarTabView as private if we intend it to be private later? Deprecating something is difficult and painful and since this is being done in the open since anyone can take a dependency on it

Reply:
At the least we should document this on SideBarTab View in the form of comments

Reply:
Yes, we will need to provide a view controller on top of it.
I spoke to our designer Today and there are still many gray areas when it comes to how this side bar will be used and where it will appear depending on screen size. 
The idea here is to have a master view controller that places all the subviews in the right place: header view, side tab bar view, tab bar view. The view controller would also take care of swapping UIViewControllers based on the selected tab.
For now, I'll try to make this SideBarTabView private. Hopefully that doesn't prevent us from using it in the demo app.

Reply:
Making SideBarTabView internal prevents us from using it in the test app. 
Is there a way to get around this limitation?
Do we really want to prevent SideBarTabView from being used outside of the intended view controller? I don't see any harm in giving more flexibility for how this view will be used.

Reply:
Yeah probably not much practical harm. The alternative is creating a trivial SideTabBarViewController wrapper around it for now, not trying to reconcile TabBarView yet and we can do that step later as you mentioned

Reply:
I will start working on the view controller in the near future (next week). 
By then, I'll be able to set SideBarTabView to internal and a temporary SideTabBarViewController won't be needed. 
I understand that this is open source and we need to be careful and intentional about what we expose but I don't think that this intermediate step is valuable, even short term.
We want to control how side tab bar and tab bar view are displayed and we can only do that with a full-fledge view controller that manages these views.